controller/src/main/java/com/epam/esm/controller/TagController.java

```diff
@@ -0,0 +1,72 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dto.TagDto;
+import com.epam.esm.linkbuilder.LinkBuilder;
+import com.epam.esm.service.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * @author Ivan Velichko
+ * @date 25.10.2021 23:34
+ */
+
+@RestController
+@Validated
+@RequestMapping("/v1/tags")
+public class TagController {
+    private final TagService tagService;
+    private final LinkBuilder<TagDto> linkBuilder;
+
+    @Autowired
+    public TagController(TagService tagService, LinkBuilder<TagDto> linkBuilder) {
+        this.tagService = tagService;
+        this.linkBuilder = linkBuilder;
+    }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    public List<TagDto> findAll(@RequestParam Map<String, String> pageParams) {
+        return tagService.findAll(pageParams).stream().peek(linkBuilder::addLinks).collect(Collectors.toList());
```

Comment:
Usually, each stream operator is written from the new row

Reply:
done)